core/src/main/scala/kafka/api/LeaderAndIsr.scala

```diff
@@ -0,0 +1,93 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package kafka.api
+
+import java.nio._
+
+import kafka.controller.LeaderIsrAndControllerEpoch
+import kafka.utils._
+
+import scala.collection.Set
+
+object LeaderAndIsr {
```

Comment:
This file is not actually new, Its just the parts that need to remain from LeaderAndIsrRequest.scala. I renamed it since the LeaderAndIsrRequest class and objects were actually removed. LeaderAndIsr and PartitionStateInfo can be removed when the last response/request using it is removed.